### PR TITLE
Optionally set -race for integration tests

### PR DESCRIPTION
**Pull request by ryanhall07**
This will be used by m3db to pass -race for some integration tests to
check for data races.

RACE=true make test-ci-integration-dbnode

### CODE REVIEW

File: test-integration.sh [hunk 1]
```diff
@@ -35,7 +36,7 @@ then
 fi
 
 # compile the integration test binary
-go test -test.c -test.tags=${TAGS} -test.covermode ${COVERMODE} \
+go test ${TEST_OPTS} -test.c -test.tags=${TAGS} -test.covermode ${COVERMODE} \
```
Comment (schallert) [inline test-integration.sh:39]: Can you please use the same manner of constructing tags used in `process-cover-package.sh`? https://github.com/m3db/ci-scripts/blob/ec7ca2ce8dfbf89017ea9f7bff7be819e2279eb1/process-cover-package.sh#L10-L15

Passing opts as a string like this is more error-prone to bad args. I know we don't have shellcheck in this repo so there's no way for linting to catch.

Comment (ryanhall07) [reply test-integration.sh:39]: that pattern doesn't really work for this. instead i'll just make an explicit `RACE=true` var and then use it to build up the TEST_OPTS internally. a similar pattern can be used in the future for more opts

**Review by robskillington — APPROVED** [review]
LGTM

File: test-integration.sh [hunk 2]
```diff
@@ -11,6 +11,8 @@ INTEGRATION_TIMEOUT=${INTEGRATION_TIMEOUT:-10m}
 COVERMODE=atomic
 SCRATCH_FILE=${COVERFILE}.tmp
 SRC_ROOT=${SRC_ROOT:-.}
+RACE=${RACE:-""}
+TEST_OPTS=""
```
Comment (schallert) [inline test-integration.sh:15]:
```suggestion
TEST_OPTS=()
```

File: test-integration.sh [hunk 3]
```diff
@@ -34,8 +36,12 @@ then
   mv coverage_imports_file.out ${DUMMY_FILE_PATH}
 fi
 
+if [ -n "${RACE}" ]; then
+  TEST_OPTS="${TEST_OPTS} -race"
```
Comment (schallert) [inline test-integration.sh:40]:
```suggestion
  TEST_OPTS+=("-race")
```

File: test-integration.sh [hunk 4]
```diff
@@ -34,8 +36,12 @@ then
...
+fi
+
 # compile the integration test binary
-go test -test.c -test.tags=${TAGS} -test.covermode ${COVERMODE} \
+go test ${TEST_OPTS} -test.c -test.tags=${TAGS} -test.covermode ${COVERMODE} \
```
Comment (schallert) [inline test-integration.sh:44]:
```suggestion
go test "${TEST_OPTS[@]}" -test.c -test.tags=${TAGS} -test.covermode ${COVERMODE} \
```

**Review by schallert — APPROVED** [review]
LGTM either way

**Pull request merged**